### ISSUE TITLE
Quote block having issue when font-family is applied that is not applied to citation.

### ISSUE THREAD

**Issue by NidhiDhandhukiya74** [WordPress/gutenberg#66794]
### Description

When we apply any font-family to quote block the pargaraph will take that font-family but citation.

### Step-by-step reproduction instructions

1. Take Quote block.
2. Add some text.
3. Use citation.
4. Now change the font family of whole quote block.

You can able to see the difference in quote paragraph text but the citation text font-family may looks the same.

### Environment info

- Wordpress 6.6.2

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [x] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by carolinan** [comment]
Hi
I am not able to reproduce this using 6.7 RC 4. I tried four different themes.
There may be styling in your theme that is affecting the citation.

**Comment by himanshupathak95** [comment]
Hi, Thanks for bringing this up.

I tested according to the instructions provided, and I was able to reproduce the same issue using WordPress `6.6.2` installation. 

Here is the testing demo  - 

https://github.com/user-attachments/assets/31270fc6-8f42-42ad-b675-e2902d1fa61b

**Comment by carolinan** [comment]
Please include the information about which themes you have tested.

**Comment by Infinite-Null** [comment]
Hi, Thanks for bringing this up.
I tested according to the instructions provided, and I was able to reproduce the same issue

https://github.com/user-attachments/assets/163ceaa3-56ac-4871-8b19-a903ef767fb3

### Testing Environment:
WordPress: 6.6.2
Theme: Twenty Twenty-Four v1.3
Browser: Chrome

**Comment by carolinan** [comment]
The reason why I am asking for the theme name is because there may be a conflict between styles added in theme.json and the block settings. That would explain why we are getting different results in different themes.

I can't reproduce it in Twenty Twenty-Three, which does not add a font family to the cite in theme.json.
I can reproduce it in Twenty-Twenty-Four, which adds a font family in theme.json. This font is part of the theme design:

```
"core/quote": {
				"elements": {
					"cite": {
						"typography": {
							"fontFamily": "var(--wp--preset--font-family--body)",
						}
					}
				},
```

In the editor, the CSS for the elements > cite is

```
:root :where(.wp-block-quote cite) {
    font-family: var(--wp--preset--font-family--body);
}
```
and it overrides the CSS for the quote block setting.
They have the same specificity, so the order will matter.

This class name, .has-heading-font-family,  is only added to the parent, the quote block:

```
.has-heading-font-family {
    font-family: var(--wp--preset--font-family--heading) !important;
}
```

And this is the CSS added by the font family setting for the quote block in theme.json:
```
:root :where(.wp-block-quote) {
    font-family: var(--wp--preset--font-family--heading);
```

**Comment by carolinan** [comment]
I am not sure if this is the best option, but the only thing I can think of would be to add the style related class names to the caption element, here:
https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/quote/edit.js#L107

**Comment by rinkalpagdar** [comment]
Hello @carolinan
A dedicated className is already assigned to the citation element: "wp-block-quote__citation".

**Comment by carolinan** [comment]
I understand. I will try to explain in more detail.

With "style related class names" I mean that when a user adds the font family, WordPress adds a class name to the block:
`has-name-font-family`.
`<blockquote class="wp-block-quote has-fira-code-font-family is-layout-flow wp-block-quote-is-layout-flow">`
Then it uses that class name to add the CSS:

```
.has-fira-code-font-family {
    font-family: var(--wp--preset--font-family--fira-code) !important;
}
```

In cases when the cite has no styling, like in Twenty Twenty-Five, the cite will also have this font family, because the element is inside the blockquote.
But when the cite is styled, the CSS added in theme.json is more specific, so the setting on the block is not used.

So my question was: If `has-name-font-family`. was on both the cite and the quote, wouldn't the font family work for both?

**Comment by rinkalpagdar** [comment]
@carolinan, I think this has been resolved based on what you suggested. It's working fine for me.
<img width="1433" alt="Image" src="https://github.com/user-attachments/assets/01b67703-ff5f-439f-860d-b6d2538091b1" />

**Comment by carolinan** [comment]
@rinkalpagdar Does it also work if the cite is styled in theme.json?

I am still seeing this:
> In cases when the cite has no styling, like in Twenty Twenty-Five, the cite will also have this font family, because the element is inside the blockquote.
> But when the cite is styled, the CSS added in theme.json is more specific, so the setting on the block is not used.

**Comment by rinkalpagdar** [comment]
Yes it's working for me even when cite has a font-family setup in theme.json

**Comment by carolinan** [comment]
Then I don't know what may be causing the different test results 🤔

**Comment by t-hamano** [comment]
I think the key point is probably this: "If a style is applied to an element inside a block via theme.json, should the style applied to the block itself override the style of that element?"

In my opinion, the answer is "it shouldn't."

For example, if you apply a red text color to a Paragraph block inside a Group block. If you then apply a blue text color to the Group block, the text color of the Paragraph block will remain red.

In other words, the idea is that styles of child blocks and child elements should always take precedence.

What do you think?

**Comment by carolinan** [comment]
I think it shouldn't.

I think it is still confusing for users though... so it comes back to not being able to see, in the interface, where a style is from, and how to override that.

**Comment by t-hamano** [comment]
> I think it is still confusing for users though... so it comes back to not being able to see, in the interface, where a style is from, and how to override that.

I agree with you on this, but it may be an issue with the global styles or the block sidebar itself.
